src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx

```diff
@@ -84,55 +87,59 @@ const getHasEnoughFundsForGas = (quote: Quote, gasFee: string, nativeNetworkAsse
   return lessThanOrEqualToWorklet(totalNativeSpentInTx, userBalance);
 };
 
-const BUFFER_FACTOR = 1.3;
-function updateMaxSwappableAmount(internalSelectedInputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>, gasFee: string) {
-  internalSelectedInputAsset.modify(asset => {
-    'worklet';
-
-    if (!asset?.isNativeAsset) return asset;
-
-    const gasFeeNativeCurrency = divWorklet(gasFee, powWorklet(10, asset.decimals));
-    const gasFeeWithBuffer = toFixedWorklet(mulWorklet(gasFeeNativeCurrency, BUFFER_FACTOR), asset.decimals);
-    const maxSwappableAmount = subWorklet(asset.balance.amount, gasFeeWithBuffer);
-
-    return {
-      ...asset,
-      maxSwappableAmount: lessThanWorklet(maxSwappableAmount, 0) ? '0' : maxSwappableAmount,
-    };
-  });
-}
-
 export function SyncGasStateToSharedValues() {
-  const { hasEnoughFundsForGas, internalSelectedInputAsset, SwapInputController } = useSwapContext();
+  const { hasEnoughFundsForGas, internalSelectedInputAsset } = useSwapContext();
 
   const { assetToSell, chainId = ChainId.mainnet, quote } = useSyncedSwapQuoteStore();
 
   const gasSettings = useSelectedGas(chainId);
   const { data: userNativeNetworkAsset } = useUserNativeNetworkAsset(chainId);
-  const { data: estimatedGasLimit, isFetching } = useSwapEstimatedGasLimit({ chainId, assetToSell, quote });
+  const { data: estimatedGasLimit } = useSwapEstimatedGasLimit({ chainId, assetToSell, quote });
+
+  const gasFeeRange = useRef<[string, string] | null>(null);
+
+  // reset gas fee range when input chain changes
+  useEffect(() => {
+    gasFeeRange.current = null;
+  }, [chainId]);
 
   useEffect(() => {
     hasEnoughFundsForGas.value = undefined;
-    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote) return;
+    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || !assetToSell) return;
 
     const gasFee = calculateGasFee(gasSettings, estimatedGasLimit);
 
-    updateMaxSwappableAmount(internalSelectedInputAsset, gasFee);
+    const nativeGasFee = divWorklet(gasFee, powWorklet(10, assetToSell.decimals));
+
+    const isEstimateOutsideRange =
+      gasFeeRange.current &&
+      (lessThanWorklet(nativeGasFee, gasFeeRange.current[0]) || greaterThanWorklet(nativeGasFee, gasFeeRange.current[1]));
+
+    // If the gas fee range hasn't been set or the estimated fee is outside the range, calculate the range based on the gas fee
+    if (nativeGasFee && (!gasFeeRange.current || isEstimateOutsideRange)) {
+      const lowerBound = toFixedWorklet(mulWorklet(nativeGasFee, LOWER_BOUND_FACTOR), assetToSell.decimals);
+      const upperBound = toFixedWorklet(mulWorklet(nativeGasFee, UPPER_BOUND_FACTOR), assetToSell.decimals);
+      gasFeeRange.current = [lowerBound, upperBound];
```

Comment:
do we need this if assetToSell is not native? could do this work after the isNativeAsset condition?

Reply:
No but I think it's good to keep it like this, so if the user switches from a non native to the native asset then there will already be a gas fee range stored and we can immediately update maxSwappableAmount